components/Account/Account.jsx

```diff
@@ -0,0 +1,118 @@
+import { useState, useEffect } from "react"
+import { useUser, useSupabaseClient } from "@supabase/auth-helpers-react"
+import styles from "./Account.module.css"
+import cn from "classnames"
+import Avatar from "@/components/Avatar/Avatar"
+
+export default function Account({ session }) {
+  const supabase = useSupabaseClient()
+  const user = useUser()
+  const [loading, setLoading] = useState(true)
+  const [name, setName] = useState(null)
+  const [avatarUrl, setAvatarUrl] = useState(null)
+
+  useEffect(() => {
+    async function getProfile() {
+      try {
+        setLoading(true)
+
+        let { data, error, status } = await supabase
+          .from("profiles")
+          .select(`*`)
+          .eq("id", user.id)
+          .single()
+
+        if (error && status !== 406) {
+          throw error
+        }
+
+        if (data) {
+          setName(data.name)
+          setAvatarUrl(data.avatar)
+        }
+      } catch (error) {
+        alert("Error loading user data!")
+        console.log(error)
+      } finally {
+        setLoading(false)
+      }
+    }
+
+    getProfile()
+  }, [session, supabase, user.id])
+
+  async function updateProfile({ name, avatar }) {
+    try {
+      setLoading(true)
+
+      const updates = {
+        id: user.id,
+        name,
+        avatar,
+        // updated_at: new Date().toISOString(),
+      }
+
+      let { error } = await supabase.from("profiles").upsert(updates)
+      if (error) throw error
```

Comment:
Currently throws an error as commented in the PR description. Violates RLS as seen in the error below.

```
{code: '42501', details: null, hint: null, message: 'new row violates row-level security policy for table "profiles"'}
```